File: request.py
#!/usr/bin/python

import cgi, cgitb, httplib, json
cgitb.enable()

print ''
q = cgi.FieldStorage()

connect = httplib.HTTPConnection('api-lenta.yandex.ru')
if q['method'].value == 'PUT':
    header = {'AUTHORIZATION': 'OAuth ' + q['token'].value, 'Content-Type': 'application/x-yandex-lenta+xml; type=read'}
else:
    header = {'AUTHORIZATION': 'OAuth ' + q['token'].value}
connect.request(q['method'].value, q['url'].value, q['data'].value, header)
resp = connect.getresponse()
result = {}
result['data'] = resp.read()
result['status'] = resp.status
result['reason'] = resp.reason
print json.dumps(result)
connect.close()

File: request.php
<?php

$token = $_GET['token'];
$url = $_GET['url'];
$method = $_GET['method'];
$data = $_GET['data'];
if ($url[0] == '/') {
    $crl = curl_init('http://api-lenta.yandex.ru'.$url);
}
else {
    $crl = curl_init($url);
}
switch ($method) {
    case 'GET' : {
        $header = array('AUTHORIZATION: OAuth '.$token);
        break;
    }
    case 'POST' :
    case 'PUT' : {
        $header = array('AUTHORIZATION: OAuth '.$token, 'Content-Type: application/x-yandex-lenta+xml; type=read');
        curl_setopt($crl, CURLOPT_POSTFIELDS, $data);
        break;
    }
}
curl_setopt($crl, CURLOPT_SSL_VERIFYPEER, 0);
curl_setopt($crl, CURLOPT_HTTPHEADER, $header);
curl_setopt($crl, CURLOPT_CUSTOMREQUEST, $method);
curl_setopt($crl, CURLOPT_RETURNTRANSFER, 1);
$result['data'] = curl_exec($crl);
$result['reason'] = curl_error($crl);
$result['status'] = curl_getinfo($crl, CURLINFO_HTTP_CODE);
echo json_encode($result);
curl_close($crl);
?>